c3/qiskit/c3_backend.py

```diff
@@ -0,0 +1,368 @@
+            "backend_name": self.name(),
+            "backend_version": self._configuration.backend_version,
+            "qobj_id": qobj.qobj_id,
+            "job_id": job_id,
+            "results": result_list,
+            "status": "COMPLETED",
+            "success": True,
+            "time_taken": (end - start),
+            "header": qobj.header.to_dict(),
+        }
+
+        return Result.from_dict(result)
+
+    def run_experiment(self, experiment: QasmQobjExperiment) -> Dict[str, Any]:
+        """Run an experiment (circuit) and return a single experiment result
+
+        Parameters
+        ----------
+        experiment : QasmQobjExperiment
+            experiment from qobj experiments list
+
+        Returns
+        -------
+        Dict[str, Any]
+            A result dictionary which looks something like::
+
+            {
+            "name": name of this experiment (obtained from qobj.experiment header)
+            "seed": random seed used for simulation
+            "shots": number of shots used in the simulation
+            "data":
+                {
+                "counts": {'0x9: 5, ...},
+                "memory": ['0x9', '0xF', '0x1D', ..., '0x9']
+                },
+            "status": status string for the simulation
+            "success": boolean
+            "time_taken": simulation time of this single experiment
+            }
+
+        Raises
+        ------
+        C3QiskitError
+            If an error occured
+        """
+        start = time.time()
+
+        # setup C3 Experiment
+        exp = Experiment()
+        exp.quick_setup(self._device_config)
+        pmap = exp.pmap
+        model = pmap.model
+
+        # initialise parameters
+        # TODO get n_qubits from device config and raise error if mismatch
+        self._number_of_qubits = experiment.config.n_qubits
+        # TODO ensure number of quantum and classical bits is same
+        shots = self._shots
+        # TODO get number of hilbert dimensions from device config
+        self._number_of_levels = 2
+
+        # Validate the dimension of initial statevector if set
+        self._validate_initial_statevector()
+
+        # TODO set simulator seed, check qiskit python qasm simulator
+        # qiskit-terra/qiskit/providers/basicaer/qasm_simulator.py
+        seed_simulator = 2441129
+
+        # TODO check for user-defined perfect and physics based sim
+        # TODO resolve use of evaluate(), process() in Experiment, possibly extend
+        # TODO implement get_perfect_gates() in Experiment
+        # unitaries = exp.get_perfect_gates()
+        exp.get_gates()
+        dUs = exp.dUs
+
+        # convert qasm instruction set to c3 sequence
+        sequence = get_sequence(experiment.instructions)
+
+        # TODO Implement extracting n_qubits and n_levels from device
+        # create initial state for qubits and levels
+        psi_init = get_init_ground_state(self._number_of_qubits, self._number_of_levels)
```

Comment:
This seems to be copy paste code from 
https://github.com/q-optimize/c3/blob/e7f4f9cf9380b1d6d0777b52c148e792b6154c90/c3/experiment.py#L147
Is there a reason not to use that method?

Reply:
Yes. Firstly, plan to replace this with a proper sequence evaluation routine in next PR. Also, related #17, I am not entirely sure of the usage of `evaluate()`, as in which all methods need to be called before invoking this. An example code snippet would be useful.

Reply:
Related #18: Using `evaluate()` requires the `tasks` to be set in the model. Currently `read_config()` does not parse for tasks, As such, using `evaluate()` would requires a hacky post-initialisation inclusion of tasks in the `Model` object which is outside the current pipeline of reading all device details from the config and hjson files. Hence, this PR won't use `evaluate()`. Once #18 is resolved, we will fix the `c3-qiskit` backend to use `evaluate()` and `process()` instead of the present implementation.